test/test_stochastic_rounding.py

```diff
@@ -0,0 +1,24 @@
+import math
+
+import torch
+import pytest
+
+
+N = 2 ** 14
+
+
+@pytest.mark.parametrize('scale', tuple(range(-18, 11)))
+def test_rs(scale):
+
+    base = math.pow(2, scale)
+    original_value = (base + math.pow(2, scale + 1)) / 2.0 + .5 * base
+    x = torch.tensor([original_value] * N).cuda()
+    _, exponent = math.frexp(original_value)
+    exponent -= 1
+    rounded = torch.stochastic_rounding(x)
+
+    mean = torch.mean(rounded).item()
+    delta_fp16 = math.pow(2, -10 + exponent if exponent >= -14 else -24)
+    threshold = 1e-6
+    diff = math.fabs(original_value - mean)
+    assert diff < threshold or diff < delta_fp16 / 2.0
```

Comment:
Could you add some comments here, please (which value for denormals etc.)?

Reply:
Added note https://github.com/csarofeen/pytorch/pull/17/files#diff-02fa1f268871375277da2c690a676146R25-R28.